community/kernel/src/main/java/org/neo4j/kernel/api/index/util/FailureStorage.java

```diff
@@ -123,6 +124,18 @@ public synchronized void storeIndexFailure( long indexId, String failure ) throw
             channel.force( true );
             channel.close();
         }
+        catch ( FileNotFoundException fnf )
```

Comment:
This was an annoying thing. This doesn't happen in reality, since in reality everything is stored in the same filesystem, but in tests the lucene in-memory filesystem is a different one from the neo in-memory filesystem, and here we are tying to access a file using the lucene paths in the neo filesystem, and that is when it all becomes a big mess. One alternative would be to change the offending test to use a real filesystem rather than an in-memory one.